.github/workflows/kiali-ci.yml

```diff
@@ -7,194 +7,24 @@ on:
-
-      - name: Upload go binary
-        uses: actions/upload-artifact@v3
-        with:
-          name: kiali
-          path: ~/go/bin/kiali
-
+    uses: ./.github/workflows/build-backend.yml
   build_frontend:
-    name: Build and unit test frontend
-    runs-on: ubuntu-20.04
-    steps:
-      - name: Checkout code
-        uses: actions/checkout@v2
-
-      - name: Setup node
-        uses: actions/setup-node@v2
-        with:
-          node-version: "14"
-          cache: yarn
-          cache-dependency-path: frontend/yarn.lock
-
-      - name: Build
-        run: make clean-all build-ui
-
-      - name: Test frontend
-        run: |
-          cd frontend
-          yarn pretty-quick --check --verbose --branch ${{ env.TARGET_BRANCH }}
-          yarn test
-
-      - name: Upload frontend build
-        uses: actions/upload-artifact@v3
-        with:
-          name: build
-          path: frontend/build/
-
-  integration_tests_frontend:
-    name: Cypress integration tests
-    needs: [build_backend, build_frontend]
-    runs-on: ubuntu-20.04
-    steps:
-      - name: Check out code
-        uses: actions/checkout@v2
-
-      - name: Setup node
-        uses: actions/setup-node@v2
-        with:
-          node-version: "14"
-          cache: yarn
-          cache-dependency-path: frontend/yarn.lock
-
-      - name: Download go binary
-        uses: actions/download-artifact@v3
-        with:
-          name: kiali
-          path: ~/go/bin/
-
-      - name: Ensure kiali binary is executable
-        run: chmod +x ~/go/bin/kiali
-
-      - name: Download frontend build
-        uses: actions/download-artifact@v3
-        with:
-          name: build
-          path: frontend/build
-
-      - name: Setup kind
-        run: hack/setup-kind-in-ci.sh
-
-      - name: Set kiali URL
-        run: |
-          KIALI_URL="http://$(kubectl get svc kiali -n istio-system -o=jsonpath='{.status.loadBalancer.ingress[0].ip}'):20001/kiali"
-          echo "::set-output name=kiali_url::$KIALI_URL"
-        id: set-kiali-url
-
-      - name: Run cypress integration tests
-        uses: cypress-io/github-action@v2
-        with:
-          working-directory: frontend
-          command: yarn run cypress:run
-        env:
-          CYPRESS_BASE_URL: ${{ steps.set-kiali-url.outputs.kiali_url }}
-          CYPRESS_NUM_TESTS_KEPT_IN_MEMORY: 0
-          CYPRESS_VIDEO: false
-          CYPRESS_SCREENSHOT_ON_RUN_FAILURE: false
-
+    uses: ./.github/workflows/build-frontend.yml
+    with:
+      # This workflow can run on either a PR or release. GITHUB_BASE_REF is only
+      # set when this workflow is run against a PR. When run against a PR, we want
+      # to use the branch we are running against: GITHUB_BASE_REF. When run
+      # as a release, we want to use the current branch ref: GITHUB_REF_NAME.
+      target_branch: ${{ github.base_ref || github.ref_name }}
   integration_tests_backend:
-    name: Backend API integration tests
```

Comment:
nit: can you keep some name for these? imo looks better with human readable names:
![Screenshot from 2022-04-06 15-57-11](https://user-images.githubusercontent.com/6226732/162060299-9d78fde8-6bd7-4e7d-99c3-98ecdb0b7e22.png)

Reply:
Sure.